pulsar-client-cpp/lib/PartitionedProducerImpl.cc

```diff
@@ -209,7 +236,10 @@ void PartitionedProducerImpl::closeAsync(CloseCallback closeCallback) {
     // Here we don't need `producersMutex` to protect `producers_`, because `producers_` can only be increased
     // when `state_` is Ready
     for (auto& producer : producers_) {
-        if (!producer->isClosed()) {
+        if (!producer->isStarted()) {
```

Comment:
what happens if a client does a sendAsync concurrently with closeAsync?

Reply:
I am working on some changes to ensure these can be called concurrently and sane things happen.